include/drivers/touchscreen.h

```diff
@@ -87,4 +87,63 @@ static inline void touchscreen_set_callback(struct device *dev,
 	api->set_callback(dev, callback);
 }
 
+
+#ifdef CONFIG_TOUCHSCREEN_HELPERS
+
+/** X,Y coordinates for a point on the touchscreen or display. */
+struct touchscreen_point {
+	u16_t x;
+	u16_t y;
+};
+
+/**
+ * @brief Touchscreen coordinate translation parameters
+ *
+ * These are initialised by calling touchscreen_set_calibration().
+ * Members should be considered private and not accessed directly.
+ */
+struct touchscreen_xlat {
+	s64_t	A;
+	s64_t	B;
+	s64_t	C;
+	s64_t	D;
+	s64_t	E;
+	s64_t	F;
+};
+
+/**
+ * @brief Initialise touchscreen coordinate translation parameters.
+ *
+ * This requires that three different and valid display/touchscreen coordinate
+ * pairs are know.
+ *
+ * @param xlat		Translation parameters to be initialised.
+ * @param display	Three display coordinates that correspond to the given
+ *			touchscreen coordinates.
+ * @param touchscreen	Three touchscreen coordinates that correspond to the
+ *			given display coordinates.
+ */
+void touchscreen_set_calibration(struct touchscreen_xlat *xlat,
+				 const struct touchscreen_point display[3],
+				 const struct touchscreen_point touchscreen[3]);
+
+/**
+ * @brief Translate touchscreen coordinates into screen coordinates.
+ *
+ * Convert a touchscreen X,Y sample into the corresponding display X,Y
+ * coordinates by using the previously initialised translation parameters.
+ *
+ * @param xlat		Translation parameters.
+ * @param dislpay_x	Display X coordinate
+ * @param dislpay_x	Display Y coordinate
+ * @param touch_x	Touchscreen X sample
+ * @param touch_y	Touchscreen Y sample
+ */
+void touchscreen_translate(struct touchscreen_xlat *xlat,
```

Comment:
And here the order of arguments is: input, output, output, input, input. Is this natural and gives a user the least surprise?